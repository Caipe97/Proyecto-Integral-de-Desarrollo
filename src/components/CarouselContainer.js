import {React, } from 'react';
//import Carousel from 'react-bootstrap/Carousel';
import MealModal from './MealModal';
import './CarouselContainer.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
const responsive = {
  superLargeDesktop: {
    // the naming can be any, depends on you.
    breakpoint: { max: 4000, min: 3000 },
    items: 6
  },
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 4
  },
  tablet: {
    breakpoint: { max: 1024, min: 464 },
    items: 3
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 2
  }
};

   

const CarouselContainer = (props) => {
    return (
        <Carousel responsive={responsive}>
        {props.meals.map((meal, index) =>
            <div className="card-container" key={index}>
                <div style={{ background: '#CCDDE2', textAlign: 'center', margin: 5, borderRadius: 10, paddingBottom: 5}}>
                    <div className="img" style={{fontSize: 24}}>
                        <p style={{textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap'}}>{meal.name}</p>
                    </div>
                    <div>
                        {meal.dateEaten.toString().replace('T', ' ').substring(0, 16)}
                    </div>
                    <div>
                        <MealModal meal={meal} history={props.history} onDeleteMeal={props.onDeleteMeal} onUpdateCurrentMealInState={props.onUpdateCurrentMealInState} />
                    </div>
                </div>
            </div>
         )}       
    </Carousel>
    )
}

export default CarouselContainer;