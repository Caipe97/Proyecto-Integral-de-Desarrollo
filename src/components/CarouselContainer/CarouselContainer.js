import {React, } from 'react';
//import Carousel from 'react-bootstrap/Carousel';
import GoalModal from '../Modal/GoalModal';
import './CarouselContainer.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
import { Button } from 'react-bootstrap';
import { Link } from "react-router-dom";
const responsive = {
  superLargeDesktop: {
    // the naming can be any, depends on you.
    breakpoint: { max: 4000, min: 3000 },
    items: 6
  },
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 4
  },
  tablet: {
    breakpoint: { max: 1024, min: 464 },
    items: 3
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 2
  }
};

   

const CarouselContainer = (props) => {
    return (
        <Carousel responsive={responsive}>
          
          <div className="card-container" key={'addMeta'}>
              <div style={{ background: '#CCDDE2', textAlign: 'center', margin: 5, borderRadius: 10, paddingBottom: 5}}>
              <Button  className="card-container" type="button" variant="primary" style={{marginBottom: 14, height: 50, width: "100%",backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>
              <Link to="/goals">Nueva Meta</Link>
              </Button>
              <Button  className="card-container" type="button" variant="primary" style={{height: 50, width: "100%",backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>
              Historial de Metas
              </Button>
              </div>
          </div>
        {props.goals.map((goal, index) =>
            <div className="card-container" key={index}>
                <div style={{ background: '#CCDDE2', textAlign: 'center', margin: 5, borderRadius: 10, paddingBottom: 5}}>
                    <div className="img" style={{fontSize: 24}}>
                        <p style={{textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap'}}>{goal.name}</p>
                    </div>
                    <div>
                        {goal.dateStart.toString().replace('T', ' ').substring(0, 7)}
                    </div>
                    {goal 
                    ?                     
                    <div>
                      <GoalModal goal={goal} history={props.history} onDeleteGoal={props.onDeleteGoal} onUpdateCurrentGoalInState={props.onUpdateCurrentGoalInState} />
                    </div>
                    : null
                    }

                </div>
            </div>
         )}       
    </Carousel>
    )
}

export default CarouselContainer;