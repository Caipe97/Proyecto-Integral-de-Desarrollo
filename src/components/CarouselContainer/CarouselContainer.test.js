import React from 'react';
import { shallow } from 'enzyme'; 
import CarouselContainer from './CarouselContainer';

describe('CarouselContainer tests', () => {
    const historyMock = { push: jest.fn() };
    const preventDefault = { preventDefault: jest.fn() }
    const deleteaa = {mealId:21, userId:1};
    const updateaa = {FoodList: [{quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}]};
    const mockPropsSignIn = {meals:  [
        {mealId: 21, name: 'queso', FoodList: [{quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}], dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
        {mealId: 22, name: 'carne', FoodList: [{quantity: 2, food: {foodId: 2, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}], dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
    ],preventDefault:  jest.fn()};
    const goals = [{
        goalId: 1,
        name: 'myGoal',
        dateStart: '2021-07',
        totalCalories: 5000,
        objectives: 
        [
            {
                objectiveCalories: 3000,
                foodCategoryId: 1
            },
            {
                objectiveCalories: 2000,
                foodCategoryId: 2
            },
        ]
    }]
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<CarouselContainer {...mockPropsSignIn} goals={goals} history={historyMock} onDeleteMeal={deleteaa} onUpdateCurrentMealInState={updateaa}/>);
    })

    it('expect to render CarouselContainer component', () => {
        expect(wrapper).toMatchSnapshot();
       
    })
    it('going to the goals when I click on this button', async () => {
        await wrapper.find('[type="button"]').at(0).simulate('click', preventDefault);
       // expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
        expect(historyMock.push.mock.calls[0]).toEqual(['/goals']);
    })
  
})
