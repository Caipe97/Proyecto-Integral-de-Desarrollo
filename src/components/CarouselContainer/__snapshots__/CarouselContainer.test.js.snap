// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests expect to render CarouselContainer component 1`] = `
<Carousel
  additionalTransfrom={0}
  arrows={true}
  autoPlaySpeed={3000}
  centerMode={false}
  className=""
  containerClass=""
  dotListClass=""
  draggable={true}
  focusOnSelect={false}
  infinite={false}
  itemClass=""
  keyBoardControl={true}
  minimumTouchDrag={80}
  pauseOnHover={true}
  renderArrowsWhenDisabled={false}
  renderButtonGroupOutside={false}
  renderDotsOutside={false}
  responsive={
    Object {
      "desktop": Object {
        "breakpoint": Object {
          "max": 3000,
          "min": 1024,
        },
        "items": 4,
      },
      "mobile": Object {
        "breakpoint": Object {
          "max": 464,
          "min": 0,
        },
        "items": 2,
      },
      "superLargeDesktop": Object {
        "breakpoint": Object {
          "max": 4000,
          "min": 3000,
        },
        "items": 6,
      },
      "tablet": Object {
        "breakpoint": Object {
          "max": 1024,
          "min": 464,
        },
        "items": 3,
      },
    }
  }
  showDots={false}
  sliderClass=""
  slidesToSlide={1}
  swipeable={true}
>
  <div
    className="card-container"
    key="addMeta"
  >
    <div
      style={
        Object {
          "background": "#CCDDE2",
          "borderRadius": 10,
          "margin": 5,
          "paddingBottom": 5,
          "textAlign": "center",
        }
      }
    >
      <Button
        active={false}
        className="card-container"
        disabled={false}
        style={
          Object {
            "backgroundColor": "rgb(18, 207, 90)",
            "borderColor": "rgb(18, 207, 90)",
            "height": 50,
            "marginBottom": 14,
            "width": "100%",
          }
        }
        type="button"
        variant="primary"
      >
        Nueva Meta
      </Button>
      <Button
        active={false}
        className="card-container"
        disabled={false}
        style={
          Object {
            "backgroundColor": "rgb(18, 207, 90)",
            "borderColor": "rgb(18, 207, 90)",
            "height": 50,
            "width": "100%",
          }
        }
        type="button"
        variant="primary"
      >
        Historial de Metas
      </Button>
    </div>
  </div>
  <div
    className="card-container"
    key="0"
  >
    <div
      style={
        Object {
          "background": "#CCDDE2",
          "borderRadius": 10,
          "margin": 5,
          "paddingBottom": 5,
          "textAlign": "center",
        }
      }
    >
      <div
        className="img"
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        <p
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          queso
        </p>
      </div>
      <div>
        2021-09-09 00:00
      </div>
      <div>
        <MealModal
          history={
            Object {
              "push": [MockFunction],
            }
          }
          meal={
            Object {
              "FoodList": Array [
                Object {
                  "food": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
              ],
              "dateEaten": "2021-09-09T00:00:00.000Z",
              "mealId": 21,
              "name": "queso",
              "userId": 1,
            }
          }
          onDeleteMeal={
            Object {
              "mealId": 21,
              "userId": 1,
            }
          }
          onUpdateCurrentMealInState={
            Object {
              "FoodList": Array [
                Object {
                  "food": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
              ],
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="card-container"
    key="1"
  >
    <div
      style={
        Object {
          "background": "#CCDDE2",
          "borderRadius": 10,
          "margin": 5,
          "paddingBottom": 5,
          "textAlign": "center",
        }
      }
    >
      <div
        className="img"
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        <p
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          carne
        </p>
      </div>
      <div>
        2021-09-09 00:00
      </div>
      <div>
        <MealModal
          history={
            Object {
              "push": [MockFunction],
            }
          }
          meal={
            Object {
              "FoodList": Array [
                Object {
                  "food": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 2,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 2,
                },
              ],
              "dateEaten": "2021-09-09T00:00:00.000Z",
              "mealId": 22,
              "name": "carne",
              "userId": 1,
            }
          }
          onDeleteMeal={
            Object {
              "mealId": 21,
              "userId": 1,
            }
          }
          onUpdateCurrentMealInState={
            Object {
              "FoodList": Array [
                Object {
                  "food": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
              ],
            }
          }
        />
      </div>
    </div>
  </div>
</Carousel>
`;
